lib/rmt/cli/products.rb

```diff
@@ -35,21 +38,70 @@ def list
   end
   map ls: :list
 
-  desc 'enable', 'Enable mirroring of product repositories by product ID or product string.'
+  desc 'enable TARGETS', 'Enable mirroring of product repositories by a list of product IDs or product strings.'
   option :all_modules, type: :boolean, desc: 'Enables all free modules for a product'
-  def enable(target)
-    change_product(target, true, options[:all_modules])
+  def enable(*targets)
+    change_products(targets, true, options[:all_modules])
   end
 
-  desc 'disable', 'Disable mirroring of product repositories by product ID or product string.'
-  def disable(target)
-    change_product(target, false, false)
+  desc 'disable TARGETS', 'Disable mirroring of product repositories by a list of product IDs or product strings.'
+  def disable(*targets)
+    change_products(targets, false, false)
   end
 
   protected
 
+  def change_products(targets, set_enabled, all_modules)
+    success = true
+    targets = clean_target_input(targets)
+    raise RMT::CLI::Error.new('No product ids supplied') if targets.empty?
+
+    targets.each do |target|
+      change_product(target, set_enabled, all_modules)
+    rescue ProductNotFoundException => e
+      puts e.message
+      success = false
+    end
+
+    raise RMT::CLI::Error.new("Not all products were #{set_enabled ? 'enabled' : 'disabled'}.") unless success
+  end
+
   def change_product(target, set_enabled, all_modules)
+    # This will return multiple products if 'SLES/15' was used
+    products = find_products(target)
+    raise ProductNotFoundException.new("No product found for target '#{target}'.") if products.empty?
+    puts "Found product(s) by target #{target}: #{products.map(&:friendly_name).join(', ')}."
+
+    products.each do |product|
+      puts "For #{product.friendly_name}:"
+
+      product_with_extensions = [product]
+      if set_enabled
+        extensions = all_modules ? Product.free_and_recommended_modules(product.id).to_a : Product.recommended_extensions(product.id).to_a
+        unless extensions.empty?
+          puts 'Enabling additional extensions:'.indent(2)
```

Comment:
Actually, aren't these /modules/, not extensions? (Or free extensions, which for the customer is the same as a module).